puppet/modules/web/files/deploy-yumrepo.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+set -xe
+# This is a forced SSH command - uncomment to test locally
+set -f -- $SSH_ORIGINAL_COMMAND
+
+prepcache() {
+	if [[ -e $REPO_PATH ]]; then
+		if [[ $MERGE == true ]] || [[ $OVERWRITE == true ]] ; then
+			echo "Repo overwrite (${OVERWRITE}) or merge (${MERGE}) are disabled, but ${REPO_PATH} already exists"
+			exit 1
+		fi
+		cp -al $REPO_PATH "$REPO_INSTANCE_PATH"
+	else
+		mkdir -p $REPO_INSTANCE_PATH
+	fi
+}
+
+do_rsync() {
+	opts=(--archive --verbose --hard-links --log-file "$REPO_RSYNC_LOG")
+	if [[ $MERGE == true ]] ; then
+		opts+=('--exclude=**/repodata/')
+	else
+		opts+=('--delete')
+	fi
+	rsync "${opts[@]}" --log-file-format 'CHANGED %f' "${REPO_SOURCE_RPM}/*" "${REPO_INSTANCE_PATH}/"
+	rsync "${opts[@]}" --log-file-format 'CHANGED source/%f' "${REPO_SOURCE_SRPM}/" "${REPO_INSTANCE_PATH}/source/"
+
+	set +f
+	for d in "${REPO_INSTANCE_PATH}"/*; do
+		(
+			cd "$d"
+
+			latest=$(ls -t foreman-release-[0-9]*.rpm 2>/dev/null | head -n1)
+			if [[ -n "$latest" ]] ; then
+				ln -sf "$latest" foreman-release.rpm
+			fi
+
+			latest=$(ls -t foreman-client-release-[0-9]*.rpm 2>/dev/null | head -n1)
+			if [[ -n "$latest" ]] ; then
+				ln -sf "$latest" foreman-client-release.rpm
+			fi
+
+			if [[ $MERGE == true ]] ; then
+				createrepo --skip-symlinks --update .
+			fi
+		)
+	done
+set -f
+}
+
+replace() {
+	if [[ -e $REPO_PATH ]]; then
+		mv "${REPO_PATH}" "${REPO_INSTANCE_PATH}-previous"
```

Comment:
should the formated version `"${REPO_INSTANCE_PATH}-previous"` be saved in a var, like `$REPO_INSTANCE_PATH_PREV`? So we can change it in one place if ever needed

Reply:
Yes, I think that's a good idea. Will you implement this or shall I?

Reply:
I will.